plugins/wasmedge_opencvmini/opencvmini_func.cpp

```diff
@@ -0,0 +1,144 @@
+
+  std::vector<char> Content(Buf, Buf + BufLen);
+  cv::Mat Img = cv::imdecode(cv::InputArray(Content), cv::IMREAD_COLOR);
+
+  return Env.insertMat(Img);
+}
+
+Expect<void> WasmEdgeOpenCVMiniImshow::body(const Runtime::CallingFrame &Frame,
+                                            uint32_t WindowNamePtr,
+                                            uint32_t WindowNameLen,
+                                            uint32_t MatKey) {
+  std::string WindowName;
+
+  // Check memory instance from module.
+  auto *MemInst = Frame.getMemoryByIndex(0);
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  char *Buf = MemInst->getPointer<char *>(WindowNamePtr);
+  std::copy_n(Buf, WindowNameLen, std::back_inserter(WindowName));
+
+  if (auto Img = Env.getMat(MatKey); Img) {
+    cv::imshow(WindowName.c_str(), *Img);
+  }
+
+  return {};
+}
+
+Expect<void> WasmEdgeOpenCVMiniWaitKey::body(const Runtime::CallingFrame &,
+                                             uint32_t Delay) {
+  cv::waitKey(static_cast<int>(Delay));
+  return {};
+}
+
+Expect<uint32_t> WasmEdgeOpenCVMiniBlur::body(const Runtime::CallingFrame &,
+                                              uint32_t SrcMatKey) {
+  cv::Mat Dst;
+  if (auto Src = Env.getMat(SrcMatKey); Src) {
+    cv::blur(*Src, Dst, cv::Size(5, 5));
+  }
+  return Env.insertMat(Dst);
+}
+
+Expect<void> WasmEdgeOpenCVMiniImwrite::body(const Runtime::CallingFrame &Frame,
+                                             uint32_t TargetFileNamePtr,
+                                             uint32_t TargetFileNameLen,
+                                             uint32_t MatKey) {
+  std::string TargetFileName;
+
+  // Check memory instance from module.
+  auto *MemInst = Frame.getMemoryByIndex(0);
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  char *Buf = MemInst->getPointer<char *>(TargetFileNamePtr);
+  std::copy_n(Buf, TargetFileNameLen, std::back_inserter(TargetFileName));
+
+  if (auto Img = Env.getMat(MatKey); Img) {
+    cv::imwrite(TargetFileName.c_str(), *Img);
+  }
+
+  return {};
+}
+
+Expect<void>
+WasmEdgeOpenCVMiniImencode::body(const Runtime::CallingFrame &Frame,
+                                 uint32_t MatKey, uint32_t BufPtr,
+                                 uint32_t BufLen) {
+  auto Img = Env.getMat(MatKey);
+  if (!Img) {
+    spdlog::error("[WasmEdge-OpenCVMini] "sv
+                  "Failed to get matrix by key."sv);
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  auto *MemInst = Frame.getMemoryByIndex(0);
+  auto OutSpan = MemInst->getSpan<uchar>(BufPtr, BufLen);
+  if (unlikely(OutSpan.size() != BufLen)) {
+    spdlog::error("[WasmEdge-OpenCVMini] "sv
+                  "Failed when accessing the image target buffer memory."sv);
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  std::vector<uchar> WriteTo;
+  cv::imencode(".jpg", *Img, WriteTo);
```

Comment:
Is it possible to expose the image types, so users can decide which format they want?

Reply:
I think we just reexport this string parameter?

Reply:
done